pointcloud/basic_octree.go

```diff
@@ -182,3 +189,158 @@ func (octree *BasicOctree) CollidesWithGeometry(geom spatialmath.Geometry, thres
 	}
 	return false, errors.New("unknown octree node type")
 }
+
+// Pose returns the pose of the octree.
+func (octree *BasicOctree) Pose() spatialmath.Pose {
+	return octree.center
+}
+
+// AlmostEqual compares the octree with another geometry and checks if they are equivalent.
+func (octree *BasicOctree) AlmostEqual(geom spatialmath.Geometry) bool {
+	// TODO
+	return false
+}
+
+// Transform recursively steps through the octree and transforms it by the given pose.
+func (octree *BasicOctree) Transform(p spatialmath.Pose) spatialmath.Geometry {
```

Comment:
How often does this get called as part of motion planning? Asking because this will be an expensive operation.

Reply:
We're currently not calling it at all except in tests. @raybjork recommended I implement this just as part of completing the Geometry implementation.

Reply:
Ah great!

Reply:
This isn't entirely true; if the octree is provided in frame of something other than world (e.g. a lidar), the octree will be transformed into the world frame.

But this will be called a single-digit number of times, not many.